mas-foundation/src/main/java/com/ca/mas/foundation/DeviceMetadata.java

```diff
@@ -0,0 +1,203 @@
+                    .build();
+
+        } catch (URISyntaxException | JSONException e) {
+            Callback.onError(callback, e);
+        }
+
+        MAS.invoke(request, new MASCallback<MASResponse<JSONObject>>() {
+            @Override
+            public void onSuccess(MASResponse<JSONObject> response) {
+                Callback.onSuccess(callback, null);
+            }
+
+            @Override
+            public void onError(Throwable e) {
+                try {
+                    if (((MASException) e).getRootCause() instanceof TargetApiException){
+                        TargetApiException tException = (TargetApiException) ((MASException) e).getRootCause();
+                        int errorCode = findErrorCode(tException.getResponse());
+
+                        if (errorCode == MAG_MAX_METADATA) {
+                            Callback.onError(callback, new MASDeviceAttributeOverflowException(errorCode, e));
+                        }
+
+                        if (errorCode == MAG_ATTR_DUPLICATED) {
+                            Callback.onError(callback, new MAGException(errorCode, "Attribute duplicated"));
+                        }
+                    }
+                } catch (Exception e1) {
+                    Callback.onError(callback, e);
+                } finally {
+                    Callback.onError(callback, e);
+                }
+            }
+        });
+    }
+
+    public static void getAttribute(String name, final MASCallback<JSONObject> callback) {
+
+        String route = ENDPOINT_PATH + "/" +name;
+        MASRequest request = null;
+
+        try {
+            request = new MASRequest.MASRequestBuilder(new URI(route))
+                    .responseBody(MASResponseBody.jsonBody())
+                    .get()
+                    .build();
+        } catch (URISyntaxException e) {
+            Callback.onError(callback, e);
+        }
+
+        MAS.invoke(request, new MASCallback<MASResponse<JSONObject>>() {
+            @Override
+            public void onSuccess(MASResponse<JSONObject> response) {
+                Callback.onSuccess(callback, response.getBody().getContent());
+            }
+
+            @Override
+            public void onError(Throwable e) {
+                try {
+                    if (((MASException) e).getRootCause() instanceof TargetApiException){
+                        TargetApiException tException = (TargetApiException) ((MASException) e).getRootCause();
+                        int errorCode = findErrorCode(tException.getResponse());
+
+                        if (errorCode == MAG_ATTR_NOT_FOUND) {
+                            // - return empty object if that resource does not exist
+                            Callback.onSuccess(callback, new JSONObject());
+                        }
+                    }
+                } catch (Exception e1) {
+                    Callback.onError(callback, e);
+                } finally {
```

Comment:
Should not provide the finally block, if the return errorCode is MAG_ATTR_NOT_FOUND, it will invoke onSuccess and onError.